target/scripts/build/packages.sh

```diff
@@ -0,0 +1,118 @@
+#! /bin/bash
+
+# -eE :: exit on error (do this in functions as well)
+# -u  :: show (and exit) when using unset variables
+# -o pipefail :: exit on error in pipes
+set -eE -u -o pipefail
+
+# shellcheck source=../helpers/log.sh
+source /usr/local/bin/helpers/log.sh
+
+QUIET='-qq' ; [[ ${LOG_LEVEL} =~ ^trace$ ]] && QUIET='-y'
+
+function _pre_installation_steps
+{
+  _log 'info' 'Starting package installation'
+  _log 'debug' 'Running pre-installation steps'
+
+  _log 'trace' 'Updating package signatures'
+  apt-get "${QUIET}" update
+
+  _log 'trace' 'Installing packages that are needed early'
+  apt-get "${QUIET}" install --no-install-recommends \
+    apt-transport-https apt-utils ca-certificates \
+    curl gnupg2 ssl-cert 2>/dev/null
+
+  _log 'trace' 'Upgrading packages'
+  apt-get "${QUIET}" upgrade
+
+  _log 'trace' 'Updating CA certificates'
+  update-ca-certificates
+}
+
+function _install_postfix
+{
+  _log 'debug' 'Installing Postfix'
+
+  _log 'warn' "Applying workaround for Postfix bug (see https://github.com/docker-mailserver/docker-mailserver/issues/2023#issuecomment-855326403)"
+
+  mv /bin/hostname /bin/hostname.bak
+  echo "echo 'docker-mailserver.invalid'" >/bin/hostname
+  chmod +x /bin/hostname
+  apt-get "${QUIET}" install --no-install-recommends postfix
+  mv /bin/hostname.bak /bin/hostname
```

Comment:
Just another note. Not suitable for this PR.

BuildKit seems to allow specifying a hostname which might work via the build ENV [`BUILDKIT_SANDBOX_HOSTNAME`](https://docs.docker.com/engine/reference/builder/#buildkit-built-in-build-args). Might be a little cleaner as a workaround, but not sure if it is any better. At least with the current approach the workaround and problem package are nicely co-located together.

Reply:
Actually, I think this is suitable for this PR - after all, this PR is about the build process. I will commit your suggestion :)

I have looked at `[BUILDKIT_SANDBOX_HOSTNAME](https://docs.docker.com/engine/reference/builder/#buildkit-built-in-build-args)` as well. I think our current solution is better since we have the workaround in one place, as you said.

---

> It does work, but that ARG requires the Dockerfile adds `# syntax=docker/dockerfile:1`

`# syntax=docker/dockerfile:1` is already added by this PR :)